services/rsources/stats_collector_test.go

```diff
@@ -37,6 +38,95 @@ var _ = Describe("Using StatsCollector", Serial, func() {
 		jobStatuses = []*jobsdb.JobStatusT{}
 	})
 
+	When("only some failed records are to be stored", func() {
```

Comment:
isn't this actually a sub-case of `"there are rudder-sources jobs"`?

Reply:
yeah this test is no longer needed imo because the interface changed.
I wrote another because of `SkipFailedRecords()`.